contracts/AliasList.sol

```diff
@@ -0,0 +1,42 @@
+pragma solidity ^0.4.24;
+
+import "./utils/Ownable.sol";
+
+
+contract AliasList is Ownable {
+    struct Alias {
+        string nick;
+        string description;
+    }
+
+    mapping(address => Alias) private aliases;
+    mapping(bytes32 => bool) private existingNicks;
+
+    constructor() public {
+        owner = msg.sender;
+    }
+
+    function setAlias(string _nick, string _description) public {
+        require(keccak256(abi.encodePacked(_nick)) != keccak256(abi.encodePacked("")), "Nick cannot be empty!");
+        require(!existingNicks[keccak256(abi.encodePacked(_nick))], "Nick already exists!");
+
+        Alias storage alias = aliases[msg.sender];
+
+        existingNicks[keccak256(abi.encodePacked(_nick))] = true;
```

Comment:
you could define local variable with this calculation to avoid unneeded calculations 3 times : `keccak256(abi.encodePacked(alias.nick))`